controllers/storageoscluster/node-manager_operand.go

```diff
@@ -349,9 +367,49 @@ func NewNodeManagerOperand(
 	}
 }
 
-//  patchStringValue specifies a patch operation for a uint32.
+//  patchUInt32Value specifies a patch operation for a uint32.
 type patchUInt32Value struct {
 	Op    string `json:"op"`
 	Path  string `json:"path"`
 	Value uint32 `json:"value"`
 }
+
+func getSideCarContainerMutator(nextIndex int, image, defaultImage string, containerName string) func(*kustomizetypes.Kustomization) {
+	if image == "" {
+		image = defaultImage
+	}
+	if image == "" {
+		return nil
+	}
+
+	return func(k *kustomizetypes.Kustomization) {
+		k.Patches = append(k.Patches, generateSideCarContainerPatch(nextIndex, containerName, image))
+	}
+}
+
+// generateSideCarContainerPatch generates a sidecar container patch.
+func generateSideCarContainerPatch(nextIndex int, name string, image string) kustomizetypes.Patch {
+	return kustomizetypes.Patch{
+		Patch: fmt.Sprintf(`[{
+			"op": "add",
+			"path": "/spec/template/spec/containers/%d",
```

Comment:
I could be wrong here, but it seems that we don't add the readiness probe anywhere